src/Utility.js

```diff
@@ -0,0 +1,232 @@
+/**
+ * @type {function}
+ * @param {number}
+ * @return {number}
+ */
+export var abs = Math.abs
+
+/**
+ * @type {function}
+ * @return {number}
+ */
+export var random = Math.random
+
+/**
+ * @type {function}
+ * @return {number}
+ */
+export var now = Date.now
+
+/**
+ * @param {*}
+ * @return {(object|function)}
+ */
+export var object = Object
+
+/**
+ * @type {function}
+ * @param {*}
+ * @param {(string|number|symbol)}
+ * @return {object}
+ */
+export var descriptors = Object.getOwnPropertyDescriptors
+
+/**
+ * @type {function}
+ * @param {object}
+ * @return {Array<string>}
+ */
+export var keys = Object.keys
+
+/**
+ * @type {function}
+ * @param {*}
+ * @param {(string|number|symbol)}
+ * @param {object}
+ * @return {object}
+ */
+export var define = Object.defineProperty
+
+/**
+ * @param {(object|function)?}
+ * @param {object}
+ * @return {object}
+ */
+export var create = Object.create
+
+/**
+ * @return {(symbol|number)} where number is not a valid array index
+ */
+export var symbol = typeof Symbol === 'function' && typeof Symbol.for === 'function' ? Symbol : define(function () {
+	return -random()
+}, 'for', function (value) {
+  for (var i = 0, h = 0; i < value.length; ++i) {
+    h = ((h << 5) - h) + value.charCodeAt(i)
+  }
+
+  return -h
+})
+
+/**
+ * @param {number} value
+ * @return {number}
+ */
+export function hash (value) {
+	return -((-(value + 1)) >>> 0)
```

Comment:
This goes through the trouble of ensuring that the number is not a valid array index but small enough that it can fit in canonical small integer representations in current JS engines. Which kind of adds weight to why [the previous comment](https://github.com/thysultan/dio.js/pull/79/commits/c967c77ad2a5b778f09e0405782838c5b6df9fd1#r211154905) should be considered.

Reply:
That might fail when the value is 1, i.e 1-1 === 0. Note the argument is always expected to be be a positive number(including zero).

Reply:
The reason i avoided that was that the double `-0` cannot be represented as a small integer. Though that might just be that i've gone to far down the rabbit hole of micro optimizations. 

That said the reason the result is applied to the >>> operator is to get a number on the upper spectrum of uint32 `-1 >>> 0` which serves to also decrease the odds of it being a value that can be authored or a valid array index. The result is then negated again. So it transition as `-1 >>> 0 -> 4294967295 -> -4294967295`. The variant `return -((-value - 1) >>> 0)` could also work if we don't worry about -0 being a double.

The reasons its both the largest possible uint32 and not a valid array index(negative) is to avoid for example conflicts with an authored key that uses the index of a map function `arr.map((v, i) => <li key={i}><li>)`. Since negative numbers are non-valid array indexes this would not conflict. But even if you did organically manage to to also negate the index i.e `arr.map((v, i) => <li key={-i}><li>)`, that still wouldn't work because the number is bordering the boundaries of the maximum length an array can be in JavaScript 4,294,967,295.